### ISSUE TITLE
Change alert notification transaction format

### ISSUE THREAD

**Issue by dexX7** [OmniLayer/omnicore#148]
Currently the alert notifications don't really fit into the usual transaction format, given that it uses plain strings with a delimiter.

I was wondering, if we should switch to the standard format in the future, and use something like `CreatePayload_Alert`, as well as `interpret_Alert`, with a format such as:
```c++
std::vector<unsigned char> CreatePayload_OmniCoreAlert(uint32_t alertType, uint64_t expiryValue, uint32_t typeCheck, uint32_t verCheck, std::string alertMessage)
{
    std::vector<unsigned char> payload;
    uint16_t messageType = 65535;
    uint16_t messageVer = 65535;
    mastercore::swapByteOrder16(messageType);
    mastercore::swapByteOrder16(messageVer);
    mastercore::swapByteOrder32(alertType);
    mastercore::swapByteOrder64(expiryValue);
    mastercore::swapByteOrder32(typeCheck);
    mastercore::swapByteOrder32(verCheck);
    if (alertMessage.size() > 255) alertMessage = alertMessage.substr(0, 255);

    PUSH_BACK_BYTES(payload, messageVer);
    PUSH_BACK_BYTES(payload, messageType);
    PUSH_BACK_BYTES(payload, alertType);
    PUSH_BACK_BYTES(payload, expiryValue);
    PUSH_BACK_BYTES(payload, typeCheck);
    PUSH_BACK_BYTES(payload, verCheck);
    payload.insert(payload.end(), alertMessage.begin(), alertMessage.end());
    payload.push_back('\0');

    return payload;
}
```

Since we have a hidden RPC call to create alerts, I don't see usability as a relevant factor.

To target old clients, it would be required to send out alerts in the old format.

**Comment by zathras-crypto** [comment]
No objections from me :)  I probably should have done it this way from the start...

> To target old clients, it would be required to send out alerts in the old format.

Sure, and that should be entirely short-term, especially if we switch to the new format in 0.0.10 (since we'd only need to target 0.0.9 based clients until they start {shutting themselves down/refusing to start} due to unsupported features being live).

**Comment by dexX7** [comment]
If possible, let's try to keep the discussion out of #169.

I'm wondering: if we update the format, then how are old-format alerts handled? I assume they wouldn't pass the transaction/payload parsing, but nothing bad happpens?

**Comment by zathras-crypto** [comment]
> how are old-format alerts handled?

The alert type is unrecognized (though sanity check needs to be updated to catch alert type 0.  Relevant bits:

Old alerts:
```
2015-08-13 22:38:32 parseTransaction(block=302202, 2014-10-19 08:21:05 idx= 8); txid: a780c0a39d4c0ff02c5914418e11d6440905a60d175e888447324d8b0eee7bd2
2015-08-13 22:38:32 	------------------------------
2015-08-13 22:38:32 	         version: 65535, class B
2015-08-13 22:38:32 	            type: 65535 (ALERT)
2015-08-13 22:38:32 	      alert type: 0
2015-08-13 22:38:32 	    expiry value: 131072
2015-08-13 22:38:32 	   alert message: 
2015-08-13 22:38:32 	          sender: mvayzbj425X55kRLLPQiuCXWUED6LMP65C
2015-08-13 22:38:32 	      authorized: false
2015-08-13 22:38:32 logicMath_Alert(): rejected: sender mvayzbj425X55kRLLPQiuCXWUED6LMP65C is not authorized for alerts
2015-08-13 22:38:32 !!! interpretPacket() returned -9051 !!!
2015-08-13 22:38:32 recordTX(a780c0a39d4c0ff02c5914418e11d6440905a60d175e888447324d8b0eee7bd2, valid=NO, block= 302202, type= 65535, value= 0)
```

```
2015-08-13 22:38:32 parseTransaction(block=304717, 2014-10-27 04:06:00 idx= 4); txid: d0ac090220b02d53a0ef88ace810fe5c308418b0f7cf3fed3cd61f70fefab9b7
2015-08-13 22:38:32 	------------------------------
2015-08-13 22:38:32 	         version: 65535, class B
2015-08-13 22:38:32 	            type: 65535 (ALERT)
2015-08-13 22:38:32 	      alert type: 13107
2015-08-13 22:38:32 	    expiry value: 808530483
2015-08-13 22:38:32 	   alert message: :1234567890123:10009:This is a test alert
2015-08-13 22:38:32 	          sender: mpZATHupfCLqet5N1YL48ByCM1ZBfddbGJ
2015-08-13 22:38:32 	      authorized: true
2015-08-13 22:38:32 New alert REJECTED (alert type not recognized): 13107, 808530483, :1234567890123:10009:This is a test alert
2015-08-13 22:38:32 recordTX(d0ac090220b02d53a0ef88ace810fe5c308418b0f7cf3fed3cd61f70fefab9b7, valid=YES, block= 304717, type= 65535, value= 0)
```

```
2015-08-13 22:38:33 parseTransaction(block=309262, 2014-11-20 06:28:50 idx= 11); txid: 3dbf4420fd8fe6d0d6cd31c7309612417a9baaaeade23b79f89121ebacc321ed
2015-08-13 22:38:33 	------------------------------
2015-08-13 22:38:33 	         version: 65535, class B
2015-08-13 22:38:33 	            type: 65535 (ALERT)
2015-08-13 22:38:33 	      alert type: 13370
2015-08-13 22:38:33 	    expiry value: 859123760
2015-08-13 22:38:33 	   alert message: 00:99:0:This version does not support TX 99v0.  You must update before block 350000 to continue using Master Core
2015-08-13 22:38:33 	          sender: mpZATHupfCLqet5N1YL48ByCM1ZBfddbGJ
2015-08-13 22:38:33 	      authorized: true
2015-08-13 22:38:33 New alert REJECTED (alert type not recognized): 13370, 859123760, 00:99:0:This version does not support TX 99v0.  You must update before block 350000 to continue using Master Core
2015-08-13 22:38:33 recordTX(3dbf4420fd8fe6d0d6cd31c7309612417a9baaaeade23b79f89121ebacc321ed, valid=YES, block= 309262, type= 65535, value= 0)
```

```
2015-08-13 22:38:33 parseTransaction(block=309579, 2014-11-21 14:35:23 idx= 4); txid: becc55dd27da89e8395e7b2ca4aa46e145f3fba15ee6462d61790a68d819eb00
2015-08-13 22:38:33 	------------------------------
2015-08-13 22:38:33 	         version: 65535, class B
2015-08-13 22:38:33 	            type: 65535 (ALERT)
2015-08-13 22:38:33 	      alert type: 12602
2015-08-13 22:38:33 	    expiry value: 858798390
2015-08-13 22:38:33 	   alert message: 00:0:This info alert is not version targeted and expires at block height 309600.
2015-08-13 22:38:33 	          sender: mvbtvhJycmKYVkaxHWgpuidyqFMnf6MEXH
2015-08-13 22:38:33 	      authorized: false
2015-08-13 22:38:33 logicMath_Alert(): rejected: sender mvbtvhJycmKYVkaxHWgpuidyqFMnf6MEXH is not authorized for alerts
2015-08-13 22:38:33 !!! interpretPacket() returned -9051 !!!
2015-08-13 22:38:33 recordTX(becc55dd27da89e8395e7b2ca4aa46e145f3fba15ee6462d61790a68d819eb00, valid=NO, block= 309579, type= 65535, value= 0)
```

New:
```
2015-08-13 22:39:05 parseTransaction(block=529467, 2015-08-11 01:32:42 idx= 3); txid: f159b11cc393bbc358b57ab89b6b1d1c9c32371eb5b900bb8fb6f06215d4cd3f
2015-08-13 22:39:05 	------------------------------
2015-08-13 22:39:05 	         version: 65535, class C
2015-08-13 22:39:05 	            type: 65535 (ALERT)
2015-08-13 22:39:05 	      alert type: 1234
2015-08-13 22:39:05 	    expiry value: 999999999
2015-08-13 22:39:05 	   alert message: Unknown alert type test
2015-08-13 22:39:05 	          sender: ms1Mjyvkx3HcfTWkCUoUhtmeYcviQpq1Ze
2015-08-13 22:39:05 	      authorized: true
2015-08-13 22:39:05 New alert REJECTED (alert type not recognized): 1234, 999999999, Unknown alert type test
2015-08-13 22:39:05 recordTX(f159b11cc393bbc358b57ab89b6b1d1c9c32371eb5b900bb8fb6f06215d4cd3f, valid=YES, block= 529467, type= 65535, value= 0)
```

```
2015-08-13 22:39:05 parseTransaction(block=529482, 2015-08-11 02:55:46 idx= 3); txid: 869d6d98a1343cd1d42c5a99cc948c68a94250fd0e2de6ab774e14cf9c232bca
2015-08-13 22:39:05 	------------------------------
2015-08-13 22:39:05 	         version: 65535, class C
2015-08-13 22:39:05 	            type: 65535 (ALERT)
2015-08-13 22:39:05 	      alert type: 1
2015-08-13 22:39:05 	    expiry value: 529580
2015-08-13 22:39:05 	   alert message: Test block expiry alert
2015-08-13 22:39:05 	          sender: ms1Mjyvkx3HcfTWkCUoUhtmeYcviQpq1Ze
2015-08-13 22:39:05 	      authorized: true
2015-08-13 22:39:05 New alert added: 1, 529580, Test block expiry alert
2015-08-13 22:39:05 recordTX(869d6d98a1343cd1d42c5a99cc948c68a94250fd0e2de6ab774e14cf9c232bca, valid=YES, block= 529482, type= 65535, value= 0)
```

```
2015-08-13 22:39:05 parseTransaction(block=529482, 2015-08-11 02:55:46 idx= 4); txid: cc44299755602d5fff71001cc28e1bc10fcba046fabe89550c882379c9a541e2
2015-08-13 22:39:05 	------------------------------
2015-08-13 22:39:05 	         version: 65535, class B
2015-08-13 22:39:05 	            type: 65535 (ALERT)
2015-08-13 22:39:05 	      alert type: 1
2015-08-13 22:39:05 	    expiry value: 1439262522
2015-08-13 22:39:05 	   alert message: Test blocktime expiry alert
2015-08-13 22:39:05 	          sender: ms1Mjyvkx3HcfTWkCUoUhtmeYcviQpq1Ze
2015-08-13 22:39:05 	      authorized: true
2015-08-13 22:39:05 New alert REJECTED (block value too large): 1, 1439262522, Test blocktime expiry alert
2015-08-13 22:39:05 recordTX(cc44299755602d5fff71001cc28e1bc10fcba046fabe89550c882379c9a541e2, valid=YES, block= 529482, type= 65535, value= 0)
```

```
2015-08-13 22:39:05 parseTransaction(block=529483, 2015-08-11 03:00:10 idx= 41); txid: c99f02f6880d9756cab6f8d0072e9f6da2be13d1e9e7ae38ef6a6b59ab665f9b
2015-08-13 22:39:05 	------------------------------
2015-08-13 22:39:05 	         version: 65535, class B
2015-08-13 22:39:05 	            type: 65535 (ALERT)
2015-08-13 22:39:05 	      alert type: 2
2015-08-13 22:39:05 	    expiry value: 1439262522
2015-08-13 22:39:05 	   alert message: Test blocktime expiry alert
2015-08-13 22:39:05 	          sender: ms1Mjyvkx3HcfTWkCUoUhtmeYcviQpq1Ze
2015-08-13 22:39:05 	      authorized: true
2015-08-13 22:39:05 New alert added: 2, 1439262522, Test blocktime expiry alert
2015-08-13 22:39:05 recordTX(c99f02f6880d9756cab6f8d0072e9f6da2be13d1e9e7ae38ef6a6b59ab665f9b, valid=YES, block= 529483, type= 65535, value= 0)
```

**Comment by dexX7** [comment]
Resolved via #176.
